scrapy/core/downloader/handlers/http11.py

```diff
@@ -440,6 +441,14 @@ def _cb_bodydone(self, result, request, url):
             certificate=result["certificate"],
             ip_address=result["ip_address"],
         )
+        if result.get("failure"):
+            # This failure is not the same object that will reach the errback,
+            # so we need to temporarily store the response in the exception.
+            # It will be moved to the failure in core/scraper.py
+            failure = result["failure"]
+            failure.value.response = response
```

Comment:
I wish I could do `failure.response = response`, but the attribute gets lost somewhere and it doesn't reach the errback. I'm open to suggestions if you can think of a more elegant alternative.

Reply:
This seems to be consistent with https://docs.scrapy.org/en/latest/topics/request-response.html#using-errbacks-to-catch-exceptions-in-request-processing, so I wonder if we just document that the way to get a response in a callback is failure.value.response when StopException is received.